src/data-model-providers/codegen/CodegenDataModelProvider_Write.cpp

```diff
@@ -110,10 +112,23 @@ DataModel::ActionReturnStatus CodegenDataModelProvider::WriteAttribute(const Dat
         VerifyOrDie((*status == Status::UnsupportedEndpoint) || //
                     (*status == Status::UnsupportedCluster) ||  //
                     (*status == Status::UnsupportedAttribute));
+
+        // Check if this is an attribute that ember does not know about but is valid after all and
+        // adjust the return code. All these global attributes are `read only` hence the return
+        // of unsupported write.
+        //
+        // If the path was not valid though, keep that return code.
+        if ((*status == Protocols::InteractionModel::Status::UnsupportedAttribute) &&
+            IsSupportedGlobalAttributeNotInMetadata(request.path.mAttributeId))
```

Comment:
Followup: we should have CI for this situation....

Reply:
I think we do - this is how I found it: https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/TestBasicInformation.yaml#L235 does:

```yaml
    - label: "Write global AttributeList attribute"
      command: "writeAttribute"
      attribute: "AttributeList"
      arguments:
          value: []
      response:
          error: UNSUPPORTED_WRITE

    - label: "Write global GeneratedCommandList attribute"
      command: "writeAttribute"
      attribute: "GeneratedCommandList"
      arguments:
          value: []
      response:
          error: UNSUPPORTED_WRITE
```